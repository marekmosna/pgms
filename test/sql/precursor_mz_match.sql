\set ECHO none
BEGIN;

\i test/pgtap-core.sql
\i sql/pgms.sql

SELECT plan(3);

SELECT is(
    precurzor_mz_match(ref, query, tolerance, tolerance_type)::numeric,
    expected,
    'precurzor_mz_match should be ' || expected
) FROM (VALUES
    (100.0::float4, 101.0::float4, 0.1::float4, 'Dalton', 0.0),
    (100.0::float4, 101.0::float4, 2.0::float4, 'Dalton', 1.0),
    (600.0::float4, 600.001::float4, 2.0::float4, 'ppm', 1.0)
) v(ref, query, tolerance, tolerance_type, expected);

SELECT * FROM finish();
ROLLBACK;
