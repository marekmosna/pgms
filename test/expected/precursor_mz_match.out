\set ECHO none
1..3
ok 1 - precurzor_mz_match should be 0.0
ok 2 - precurzor_mz_match should be 1.0
ok 3 - precurzor_mz_match should be 1.0
